contributing/building-a-bento-amp-extension.md

```diff
@@ -0,0 +1,701 @@
+in a fully AMP document.
+
+The configurations which bridge the Preact implementation of the component and its custom element counterpart in an HTML or AMP document are explained in the [AMP/Preact Bridge](#amp/preact-bridge) section, and the callbacks which handle AMP- and DOM- specific mutability traits are explained in the [PreactBaseElement Callbacks](#preactbaseelement-callbacks) section. All of these are also explained inline in the [PreactBaseElement](https://github.com/ampproject/amphtml/blob/master/src/preact/base-element.js) class.
+
+### Element and Component classes
+
+The following shows the overall structure of your element implementation
+file (`extensions/amp-my-element/0.1/amp-my-element.js`). See [Experiments](#experiments) to make sure your component is experimentally gated if necessary.
+
+```js
+import {func1, func2} from '../../../src/module';
+import {MyElement} from './my-element'; // Preact component.
+import {CSS} from '../../../build/amp-my-element-0.1.css';
+// more ES2015-style import statements.
+
+/** @const */
+const EXPERIMENT = 'amp-my-element';
+
+/** @const */
+const TAG = 'amp-my-element';
+
+class AmpMyElement extends AMP.PreactBaseElement {
+  /** @override */
+  init() {
+    // Perform any additional processing of prop values that are not
+    // straightforward attribute passthroughs, as well as AMP-specific
+    // work such as registering actions and events.
+    this.registerApiAction('close', (api) => api.close());
+
+    const processedProp = parseInt(element.getAttribute('data-binary'), 2);
+    return dict({
+      'processedProp': processedProp,
+      'onClose': (event) => fireAmpEvent(event)}
+    );
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == LAYOUT.FIXED;
+  }
+}
+
+AmpMyElement['Component'] = MyElement;            // Component definition.
+
+AmpMyElement['props'] = {  // Map DOM attributes to Preact Component props.
+  'propName1': {attr: 'attr-name-1'},
+  'propName2': {attr: 'attr-name-2', type: 'number'},
+};
+
+AMP.extension('amp-my-element', '0.1', (AMP) => {
+  AMP.registerElement('amp-my-element', AmpMyElement, CSS);
+});
+```
+
+Accordingly, the following shows the corresponding overall structure of your Preact functional component implementation file (extensions/amp-my-element/0.1/my-element.js).
+
+```js
+import * as Preact from '../../../src/preact';
+import {func1, func2} from '../../../src/module';
+// more ES2015-style import statements.
+
+export function MyElement({propName1, propName2, ...rest}) {
+  // Assign state variables or other behaviors using Preact
+  // lifecycle hooks such as useRef, useState, useEffect, etc.
+  const [foo, useFoo] = useState(null);
+
+  // Render component via JSX syntax.
+  return (
+    <div>Hello world</div>
+  );
+}
+```
+
+### PreactBaseElement callbacks
+
+#### checkPropsPostMutations
```

Comment:
![image](https://user-images.githubusercontent.com/6694512/104211482-1f7ebf00-5402-11eb-976d-023cb8c0e2ed.png)
Nope! I noticed that this specific section in the parallel doc also lacked backticks, but I'm guessing it was just for convenience. Thanks for adding them here!